File: interview-preparation-kit/archive/sorting/mark_and_toys.py
import math
import os
import random
import re
import sys

# Complete the maximumToys function below.
def maximumToys(prices, k):

        affordable_prices = list(filter(lambda x: x<=k, prices))
        affordable_prices = sorted(affordable_prices) #this is hard coded? Surely you can order?

        spend = 0
        num_toys = 0
        print(affordable_prices)
        for i in affordable_prices:
            spend+=i
            if spend<=k: num_toys+=1
            else: break

        return num_toys


if __name__ == '__main__':

    nk = input().split()
    n = int(nk[0])
    k = int(nk[1])

    prices = list(map(int, input().rstrip().split()))
    result = maximumToys(prices, k)

    print(str(result) + '\n')

File: interview-preparation-kit/archive/sorting/mark_and_toys.js
'use strict';

const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({input: process.stdin, output: process.stdout});

let inputString = [];
let currentLine = 0;

rl.prompt();

rl.on('line', (inputLine) => {
    inputString.push(inputLine);
})

rl.on('close', () => {
    console.log(main(inputString));
    process.exit(0);
})

function maximumToys(prices, k) {

    const affordablePrices = prices.filter(price => price<=k)
    affordablePrices.sort((a, b) => a-b) //Weidly JS convert all array elements to strings before sort... so it doesnt quite work..

    let spend = 0;
    let numToys = 0;

    for (var i=0; i<affordablePrices.length; i++) {
        spend+=affordablePrices[i]
        if (spend<=k) {
            numToys++;
        } else {
            break
        }
    }

    return numToys;
}

function main(inputString) {
    const nk = inputString[0].split(' ');

    const n = parseInt(nk[0], 10);
    const k = parseInt(nk[1], 10);
    const prices = inputString[1].split(' ').map(i => parseInt(i, 10));

    const result = maximumToys(prices, k);

    return result+'\n'
}
